### DISCUSSION TITLE
Support & fix change suggestion in the presence of deleted lines

### DISCUSSION THREAD

**Discussion by xmo-odoo** [community/community#32114] [Pull Requests]
Currently, when selecting lines spanning deletions e.g.
```diff
+ b
- a
+ c
```
the suggestion button will not appear. Furthermore, when suggesting a change (manually) anyway, the suggestion shows up as
```diff
- b
- a
- c
+ other
```
Which is completely nonsensical, it should be either
```diff
- a
+ other
```
(the final code) or
```diff
- b
- c
+ other
```
(the suggestion being made)

The current behaviour is very inconvenient, because the diff style github uses means small removals are common. And furthermore it's not rare that a snippet which was partially being updated can now be better rewritten in its entirety.

**Comment by Dreamsorcerer** [comment]
I'm also finding that this triggers even when lines haven't been deleted. But, when I select code on the right side to comment on, it automatically selects the code on the left side as well (where a simple change is seen as a deletion on the left, and an addition on the right).

![image](https://user-images.githubusercontent.com/2203121/198092149-a2f15062-b1a5-42ac-b65e-2f327d708842.png)

**Reply by xmo-odoo** [reply]
FWIW the side by side diff view is fundamentally just a different way of displaying the unified diff, so it has the same issue, as far as github's underlying model is concerned that's a deleted line and two added lines.

If you switch to unified diff view you'll see how github "actually" sees it in their backend, and likely it sees it as the fist case, where the import line is added, then the `create_pool` line is removed, then the `from_url` line is added.

**Comment by truh** [comment]
I question why it's disabled for deletions in the first place. When I use the suggest changes feature my intention is primarily to suggest changes to the latest iteration of files in the pull request rather than to edit the diff.
